File: refactoring-practice-1/src/main/typescript/form_template_method/lifeline-site.ts
import {Site} from "./site.js";

export class LifelineSite extends Site {

    public constructor(units: number, rate: number) {
        super(units, rate);
    }

    public getBillableAmount() {
        let base = this.getUnits() * this.getRate() * 0.5;
        let tax = base * Site.TAX_RATE * 1.2;
        return base + tax;
    }

}

File: refactoring-practice-1/src/main/java/dojo/form_template_method/LifelineSite.java
package dojo.form_template_method;

public class LifelineSite extends Site {

	public LifelineSite(int units, double rate) {
		super(units, rate);
	}
	
	public double getBillableAmount() {
		double base = getUnits() * getRate() * 0.5;
		double tax = base * TAX_RATE;
		return base + tax;
	}

}
